src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -0,0 +1,265 @@
+ * roundabout is multi-directional, or where the roundabout has incorrect geometry (concave).
+ *
+ * @author savannahostrowski
+ */
+
+public class MalformedRoundaboutCheck extends BaseCheck
+{
+    private static final long serialVersionUID = -3018101860747289836L;
+    private static final String WRONG_WAY_INSTRUCTIONS = "This roundabout, {0,number,#},is going the"
+            + " wrong direction, or has been improperly tagged as a roundabout.";
+    private static final String MULTIDIRECTIONAL_INSTRUCTIONS = "This roundabout, {0,number,#}, is"
+            + " multi-directional, or the roundabout has improper angle geometry.";
+    private static final String[] LEFT_DRIVING_COUNTRIES_DEFAULT = { "AIA", "ATG", "AUS", "BGD",
+            "BHS", "BMU", "BRB", "BRN", "BTN", "BWA", "CCK", "COK", "CXR", "CYM", "CYP", "DMA",
+            "FJI", "FLK", "GBR", "GGY", "GRD", "GUY", "HKG", "IDN", "IMN", "IND", "IRL", "JAM",
+            "JEY", "JPN", "KEN", "KIR", "KNA", "LCA", "LKA", "LSO", "MAC", "MDV", "MLT", "MOZ",
+            "MSR", "MUS", "MWI", "MYS", "NAM", "NFK", "NIU", "NPL", "NRU", "NZL", "PAK", "PCN",
+            "PNG", "SGP", "SGS", "SHN", "SLB", "SUR", "SWZ", "SYC", "TCA", "THA", "TKL", "TLS",
+            "TON", "TTO", "TUV", "TZA", "UGA", "VCT", "VGB", "VIR", "WSM", "ZAF", "ZMB", "ZWE" };
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(WRONG_WAY_INSTRUCTIONS,
+            MULTIDIRECTIONAL_INSTRUCTIONS);
+
+    private Set<String> leftDrivingCountries;
+
+    /**
+     * An enum of RoundaboutDirections
+     */
+    public enum RoundaboutDirection
+    {
+        CLOCKWISE,
+        COUNTERCLOCKWISE,
+        // Handles the case where multiple directions were found in the roundabout
+        MULTIDIRECTIONAL,
+        // Handles the case where we were unable to get any information about the roundabout's
+        // Direction or if the roundabout's geometry was malformed (concave).
+        UNKNOWN
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public MalformedRoundaboutCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.leftDrivingCountries = new HashSet<>(
+                Arrays.asList((String[]) configurationValue(configuration, "traffic.countries.left",
+                        LEFT_DRIVING_COUNTRIES_DEFAULT)));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // We check that the object is an instance of Edge
+        return object instanceof Edge
+                // Make sure that the object has an iso_country_code
+                && object.getTag(ISOCountryTag.KEY).isPresent()
+                // Make sure that the edges are instances of roundabout
+                && JunctionTag.isRoundabout(object)
+                // Is not two-way
+                && !OneWayTag.isExplicitlyTwoWay(object)
```

Comment:
Just curious is it possible to have two way roundabouts? If not, it might worth flagging them as well.

Reply:
Yeah, you're right. I don't think that it's valid to have a two-way roundabout. I'll remove that.

Reply:
Cool. The craziest roundabout I know is in England: https://www.openstreetmap.org/#map=19/51.56265/-1.77127